### PR TITLE
feat: run both ucc and legacy modinput tests

### PR DESCRIPTION
**Pull request by dvarasani-crest**
This PR adds the feature to run the `ucc_modinput_functional` tests seperately from `modinput_functional` (Legacy).

Reference: https://splunk.atlassian.net/browse/ADDON-75556

Test run: 
https://github.com/splunk/splunk-add-on-for-salesforce/actions/runs/11473918131
https://github.com/splunk/splunk-add-on-for-google-cloud-platform/actions/runs/12067263557

### CODE REVIEW

File: .github/workflows/reusable-build-test-release.yml [hunk 1]
```diff
@@ -2064,6 +2066,280 @@ jobs:
...
+        run: |
+          ARGO_TOKEN=$(aws secretsmanager get-secret-value --secret-id "${{ needs.setup-workflow.outputs.argo_token_secret_id_k8s }}" | jq -r '.SecretString')
+          echo "argo-token=$ARGO_TOKEN" >> "$GITHUB_OUTPUT"
+      - name: create job name
+        id: create-job-name
+        shell: bash
+        run: |
+          RANDOM_STRING=$(head -3 /dev/urandom | tr -cd '[:lower:]' | cut -c -4)
+          JOB_NAME=${{ needs.setup.outputs.job-name }}-${RANDOM_STRING}
+          JOB_NAME=${JOB_NAME//TEST-TYPE/ucc_modinput}
```
Comment (kdoroszko-splunk) [inline .github/workflows/reusable-build-test-release.yml:2134]: Can we keep the convention from other test types?
```suggestion
          JOB_NAME=${JOB_NAME//TEST-TYPE/${{ env.TEST_TYPE }}}
```

File: .github/workflows/reusable-build-test-release.yml [hunk 2]
```diff
@@ -2064,6 +2066,280 @@ jobs:
...
+        run: |
+          echo "| Job | Total Tests | Passed Tests | Failed Tests | Errored Tests | Skipped Tests | Report Link" >> "$GITHUB_STEP_SUMMARY"
+          echo "| ---------- | ----------- | ------ | ------ | ------ | ------- | ------ |" >> "$GITHUB_STEP_SUMMARY"
+          for file in summary-ucc_modinput*/job_summary.txt; do
+            cat "$file" >> "$GITHUB_STEP_SUMMARY"
+          done
+      - uses: geekyeggo/delete-artifact@v5
+        with:
+            name: |
+                summary-modinput*
```
Comment (kdoroszko-splunk) [inline .github/workflows/reusable-build-test-release.yml:2341]: `summary-modinput*` or `summary-ucc_modinput*`
```suggestion
                summary-ucc_modinput*
```

File: .github/workflows/reusable-build-test-release.yml (hunk 1 above)
Comment (dvarasani-crest) [reply .github/workflows/reusable-build-test-release.yml:2134]: I kept it that way only earlier, but was facing issues in some case as it was exceeding the name length limit in the argo workflow.

File: .github/workflows/reusable-build-test-release.yml [hunk 3]
```diff
@@ -33,7 +39,7 @@ on:
         required: false
         description: "branch for k8s manifests to run the tests on"
         type: string
-        default: "v3.1.0"
+        default: "feat/run-both-ucc-and-legacy-modinput-tests"
```
Comment (kdoroszko-splunk) [inline .github/workflows/reusable-build-test-release.yml:42]: Let's merge this PR https://github.com/splunk/ta-automation-k8s-manifests/pull/110#pullrequestreview-2469961146 at first and put here new version

Comment (dvarasani-crest) [reply .github/workflows/reusable-build-test-release.yml]: done!

File: README.md [hunk 4]
```diff
@@ -588,6 +602,45 @@ helmut.log
...
+  To do so, one must specify `ucc-modinput-marker` parameter in buid-test-release.yml as in [example](https://github.com/splunk/splunk-add-on-for-google-cloud-platform/blob/34abcf2780d8f223f292c9c2fcc5835b71a8de99/.github/workflows/build-test-release.yml#L34).
+  Markers must be created prior and each test case must be marked (check the following references: [ref1](https://github.com/splunk/splunk-add-on-for-google-cloud-platform/blob/34abcf2780d8f223f292c9c2fcc5835b71a8de99/tests/ucc_modinput_functional/markers.py), 
+[ref2](https://github.com/splunk/splunk-add-on-for-google-cloud-platform/blob/34abcf2780d8f223f292c9c2fcc5835b71a8de99/tests/ucc_modinput_functional/test_google_cloud_rh_settings.py#L19))
+
+**Action used:** 
+- No action
+
+**Pass/fail behaviour**
+
+- The stage is expected to fail only if there are any Modular input test failures defined under tests/ucc_modular_functional
```
Comment (kdoroszko-splunk) [inline README.md:620]: Do we keep this tests in `tests/ucc_modular_functional` directory?

Comment (dvarasani-crest) [reply README.md:620]: yes.

Comment (dvarasani-crest) [reply README.md:620]: There are some TAs, which has both the legacy and new tests present and to differentiate it, they are put under different directories.

Comment (dvarasani-crest) [reply README.md]: It should be` tests/ucc_modinput_functional`. Updated!

File: README.md [hunk 5]
```diff
@@ -588,6 +602,45 @@ helmut.log
...
+  To do so, one must specify `ucc-modinput-marker` parameter in buid-test-release.yml as in [example](https://github.com/splunk/splunk-add-on-for-google-cloud-platform/blob/34abcf2780d8f223f292c9c2fcc5835b71a8de99/.github/workflows/build-test-release.yml#L34).
+  Markers must be created prior and each test case must be marked (check the following references: [ref1](https://github.com/splunk/splunk-add-on-for-google-cloud-platform/blob/34abcf2780d8f223f292c9c2fcc5835b71a8de99/tests/ucc_modinput_functional/markers.py), 
+[ref2](https://github.com/splunk/splunk-add-on-for-google-cloud-platform/blob/34abcf2780d8f223f292c9c2fcc5835b71a8de99/tests/ucc_modinput_functional/test_google_cloud_rh_settings.py#L19))
+
+**Action used:** 
+- No action
+
+**Pass/fail behaviour**
+
+- The stage is expected to fail only if there are any Modular input test failures defined under tests/ucc_modinput_functional
```
Comment (kdoroszko-splunk) [inline README.md:620]: I still have doubts here: don't we keep these tests under tests/modinput_functional, but the difference is  in centaurus directory?

Comment (dvarasani-crest) [reply README.md:620]: Some of the TAs have both the legacy and new modinput tests present at the same time. So, to differentiate them they are kept in different directory.

**Comment by dvarasani-crest** [comment]
> It seems like `splunk-add-on-ucc-modinput-test-functional.log` is being generated as an artifact for each test type (even for UI and KO). Is it possible to change that on our side?

Log file is being generated because of the hooks implemented in : [splunk_add_on_ucc_modinput_test](https://github.com/splunk/addonfactory-ucc-test/blob/gcp_ta_adoption_related_changes/splunk_add_on_ucc_modinput_test/functional/pytest_plugin/hooks.py).
In the [AWS TA](https://github.com/splunk/splunk-add-on-for-amazon-web-services/blob/test/ucc-modinput/pyproject.toml), the dependency `splunk_add_on_ucc_modinput_test` is not marked as optional in the pyproject.toml resulting in getting installed for each test type. However it was marked as optional in the [Salesforce TA](https://github.com/splunk/splunk-add-on-for-salesforce/blob/refs/heads/test/ucc-modinput/pyproject.toml#L7), and not getting installed for other test type there.

**Review by kdoroszko-splunk — APPROVED** [review]

**Pull request merged**